### PR TITLE
Refactor Artifact creation.

### PR DESCRIPTION
**Pull request by Askir**
I''ve extracted all the creation logic into a seperate file. Which will hopefully enable me to reuse that logic when downloading files from google drive, without sending actual POST requests to /images.

I also renamed the ArtifactBuilder to ArtifactConnector so it doesn't get confused with ArtifactCreator. If there are any other naming suggestions. I'm very open to them. Seems at least slightly confusing to me as well still.

### CODE REVIEW

File: application/artifacts/artifact_connector.py [hunk 1]
```diff
@@ -66,7 +64,7 @@ def update_with(self, override_tags=True, **properties):
     @classmethod
     def find(cls, uid, force=True):  # pylint: disable= invalid-name
```
Comment (arne-z) [inline application/artifacts/artifact_connector.py]: `  # pylint: disable= invalid-name`
should no longer be necessary.

File: application/artifacts/artifact_creation.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+import werkzeug
+from PIL import Image
+from flask import current_app
+
+from application.artifacts.artifact_connector import ArtifactConnector
+
+from application.artifacts.image_recognition import ImageRecognizer
+
+
+class ImageResizer:  # pylint:disable=too-few-public-methods
```
Comment (arne-z) [inline application/artifacts/artifact_creation.py]: `# pylint:disable=too-few-public-methods`
should not be necessary.

File: application/artifacts/artifact_creation.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    def save_sizes(self):
+        """Saves resized image for all defined widths"""
+        for width in self.widths:
+            height = self._calculate_height(width)
+            resized_image = self.image.resize((width, height), Image.ANTIALIAS)
+            resized_image.save(self._generate_file_path(width))
+            resized_image.close()
+
+    def _generate_file_path(self, width):
```
Comment (arne-z) [inline application/artifacts/artifact_creation.py:31]: Could add docstring here just for consistency.

File: application/artifacts/artifact_connector.py [hunk 4]
```diff
@@ -1,18 +1,16 @@
...
+from application.artifacts.property_builder import PropertyBuilder
 from application.artifacts.tags.tag import Tag
 from application.teams.team import Team
 from application.users.user import User
-from application.artifacts.property_builder import PropertyBuilder
 
 
-class ArtifactBuilder:
+class ArtifactConnector:
     """This class helps with building Artifacts"""
```
Comment (arne-z) [inline application/artifacts/artifact_connector.py]: Could make docstring more specific (explain why it's a connector).

File: application/artifacts/artifact_creation.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+
+class ArtifactCreator:
+    def __init__(self, file, owner_id, team_id, **optional):
+        self.file = file
+        self.owner_id = owner_id
+        self.team_id = team_id
+        self.additional_params = optional
+
+    def create_artifact(self):
```
Comment (arne-z) [inline application/artifacts/artifact_creation.py:94]: Docstring?

File: specs/requests/artifacts_spec.py [hunk 6]
```diff
@@ -38,7 +38,7 @@ def clear_upload_dir():
         self.context.client().login(self.user)
 
     with after.each:
-        clear_upload_dir()
+        # clear_upload_dir()
```
Comment (arne-z) [inline specs/requests/artifacts_spec.py]: This can't be right?

**Review by arne-z — CHANGES_REQUESTED** [review]

File: application/artifacts/artifact_creation.py (hunk 3 above)
Comment (Askir) [reply application/artifacts/artifact_creation.py:31]: I wont add a docstring to private methods (except if it isn't clear what the are doing by method name). But I added a class comment and one to the public method.

File: specs/requests/artifacts_spec.py (hunk 6 above)
Comment (Askir) [reply specs/requests/artifacts_spec.py]: nope, good catch! :D

**Review by Adrian-St — APPROVED** [review]
I like it :)

File: application/artifacts/artifact_creation.py (hunk 3 above)
Comment (arne-z) [reply application/artifacts/artifact_creation.py:31]: 👍

**Review by arne-z — APPROVED** [review]

**Pull request merged**